libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -262,6 +275,27 @@ proc sendEncoded*(p: PubSubPeer, msg: seq[byte]) {.async.} =
 
     await conn.close() # This will clean up the send connection
 
+proc sendEncoded*(p: PubSubPeer, msg: seq[byte], isHighPriority: bool = false) {.async.} =
```

Comment:
the `async` here introduces an additional copy of `msg` and is redundant given that all callers `asyncSpawn` the send more or less..

In general, the consequence of using `asyncSpawn` is that `await p.rpcmessagequeue.nonPriorityQueue.addLast(msg)` loses its meaning and effectively disables any limits placed on the queue (because all the spawned tasks will be stuck here).

Given that the policy of the queue is time-based (and not slot-based), it's better to make this function non-async and instead use `addLastNoWait` - it's guaranteed not to raise if the queue is unbounded.

To understand why `send` is non-blocking (not async), we need to go back to #342 and look at how it avoids a deadlock between sending and waiting for data

Reply:
We `asyncSpawn` `peer.sendEncoded` at pubsub:204